components/random_cut.py

```diff
@@ -0,0 +1,52 @@
+import numpy as np
+from heapq import heapify, heappush, heappop
+
+
+def random_cut_generator(num_bins, bin_length, bin_width, bin_height, num_cases,
+                         seed=111, **kwargs):
+    bin_height_ = bin_height / 2
+    np.random.seed(seed)
+    boxes = [(-max(bin_length, bin_width, bin_height_),
+              bin_length, bin_width, bin_height_)] * num_bins
+    heapify(boxes)
+    c = 0
+    while len(boxes) < num_cases and c < 1000:
+        b = heappop(boxes)
+        i = 1 + np.argmax(b[1:])
+        r = 0.5 + (np.random.rand() - 0.5) / 4
+        box1 = list(b)
+        box2 = list(b)
+        cut_length = int(b[i] * r)
```

Comment:
This can sometimes generate dimensions of zero for cases, for example when setting a bin dimension to 1.